### PR TITLE
PSR2/UseDeclaration: allow comments after the last USE statement

### LINKED ISSUE #1891: Comment on last USE statement causes false positive for PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse
# Overview
If the last USE statement has a comment, an incorrect `PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse` error is reported.

# Steps to reproduce
Run `phpcs --standard=PSR2` on a file with the following contents:
```php
<?php
use Foo\Bar; //comment

// This should pass.
```

## Expected output
None. This should validate correctly.

## Actual output
```
FILE: /Users/matt/Projects/PHP_CodeSniffer/tests/TestCommentAfterLastUse.php
-------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------
 2 | ERROR | [x] There must be one blank line after the last USE statement; 0 found;
-------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------

Time: 114ms; Memory: 6Mb
```

### PR DESCRIPTION
**Pull request by iammattcoleman**
Resolves #1891

### CODE REVIEW

File: src/Standards/PSR2/Sniffs/Namespaces/UseDeclarationSniff.php [hunk 1]
```diff
@@ -142,7 +142,7 @@ public function process(File $phpcsFile, $stackPtr)
             return;
         }
 
-        $next = $phpcsFile->findNext(T_WHITESPACE, ($end + 1), null, true);
+        $next = $phpcsFile->findNext([T_WHITESPACE, T_PHPCS_IGNORE, T_COMMENT], ($end + 1), null, true);
```
Comment (jrfnl) [inline src/Standards/PSR2/Sniffs/Namespaces/UseDeclarationSniff.php]: You may want to use `Tokens::$EmptyTokens` instead for completeness.

Comment (iammattcoleman) [reply src/Standards/PSR2/Sniffs/Namespaces/UseDeclarationSniff.php]: Thanks. Updated!

**Comment by gsherwood** [comment]
Thanks a lot for fixing this, and for the test.

**Comment by dhensby** [comment]
This appears to have caused a regression, please see #1898

**Comment by dhensby** [comment]
OK - the regression here is that this allows you to do:

```php
use Class; // comment
```

but it now causes this to fail:

```php
use Class;

//comment
```

Because the test fixture tried to evaluate both at once it *appeared* to fix the problem, but it actually caused a regression instead

**Pull request merged**